docs/apache-airflow-providers-salesforce/index.rst

```diff
@@ -82,28 +82,8 @@ PIP package              Version required
 ``apache-airflow``       ``>=2.1.0``
 ``pandas``               ``>=0.17.1``
 ``simple-salesforce``    ``>=1.0.0``
-``tableauserverclient``
 =======================  ==================
 
-Cross provider package dependencies
```

Comment:
Is it worth keeping a note here saying "want tableu? It's moved"?

Reply:
That is covered by the change log of the provider :) 

1. If someone is currently using Tableau via Salesforce - the user is aware it's deprecated by warnings and will know it's removed by reading the change log.
2. If someone never used Tableau before and searches for tableau provider I want him/her to land in Tableau docs rather than Salesforce docs only to be referred to Tableau provider.

Reply:
This is all good in general. We have "addional extra" - if somoene wants tableau they should install salesforce provider with `["tableau"]` extra (or install `[tableau]' extra in airflow or install `apache-airflow-providers-extra'. But I agree we should add a note in the chanelog of salesforce. And answering the question of @ashb below - this is described actually in changelog.rst of each provider (kind of preparation to make them fully independent and splittable): 

https://raw.githubusercontent.com/apache/airflow/main/airflow/providers/salesforce/CHANGELOG.rst

```
.. NOTE TO CONTRIBUTORS:
   Please, only add notes to the Changelog just below the "Changelog" header when there are some breaking changes
   and you want to add an explanation to the users on how they are supposed to deal with them.
   The changelog is updated and maintained semi-automatically by release manager.
```

Reply:
Hmm. Looking at the build image failure in the PR. looks strange.

Reply:
added note https://github.com/apache/airflow/pull/23747/commits/19f6aced38e2d76560e283447f1e111cef171be6